handler/transport/oauth2.go

```diff
@@ -40,6 +40,14 @@ func NewOAuth2(config *config.OAuth2, memStore *cache.MemoryStore,
 		return nil, fmt.Errorf("the grant_type has to be set to 'client_credentials'")
 	}
 
+	if config.Retries == nil {
```

Comment:
Using `*uint8` would give the responsibility for negative numbers to the hcl decoding.
Additionally its range is capped by type: 0 - 255 which should be enough retries.
If this is `nil` the default can be set... on config level. The specific handling here would be obsolete.

Reply:
Done